### ISSUE TITLE
Async version of forget block

### ISSUE THREAD

**Issue by atasak** [dsherret/ts-simple-ast#194]
The forget blocks work really nice, but whenever you've got an async function you've got a problem. Proposal:

```typescript
await ast.forgetNodesCreatedInBlockAsync(async () => {
    ...
}
```

Where `ast.forgetNodesCreatedInBlockAsync` expects a function that returns a Promise and returns a Promise.

**Comment by dsherret** [comment]
Yes, that would be good. Thanks for the suggestion!

**Comment by dsherret** [comment]
I managed to get this so that it's basically the same code... just add the async and await keywords:

```
await ast.forgetNodesCreatedInBlock(async () => {
    ...
});
```

**Comment by dsherret** [comment]
@atasak this is released now in v5.0 along with the bug fixes for the other issues.

**Comment by atasak** [comment]
Thanks, now I don't have to do everything synchronous :)
